src/ObservationDims.jl

```diff
@@ -1,5 +1,163 @@
 module ObservationDims
 
-greet() = print("Hello World!")
+using AxisArrays
+using Compat
+using Distributions
+using NamedDims
+
+export obs_arrangement, organise_obs
+export SingleObs, IteratorOfObs, ArraySlicesofObs
+export MatrixRowsOfObs, MatrixColsOfObs
+
+"""
+    type ObsArrangement
+
+Defines the orientation of data that is expected by a function
+"""
+abstract type ObsArrangement end
+
+struct SingleObs <: ObsArrangement end
+struct IteratorOfObs <: ObsArrangement end
+struct ArraySlicesOfObs{D} <: ObsArrangement end
+
+const MatrixRowsOfObs = ArraySlicesOfObs{1}
+const MatrixColsOfObs = ArraySlicesOfObs{2}
+
+
+"""
+    obs_arrangement(f::Function) -> ObsArrangement
```

Comment:
should this be a `Type`? like `obs_arrangement(::typeof(mean_squared_error)) = SingleObs()`
```suggestion
    obs_arrangement(f::Type{<:Function}) -> ObsArrangement
```